src/cygnus.js

```diff
@@ -1,194 +1,186 @@
-var _this2 = this,
-    _arguments = arguments;
+/* global Blob, Worker, XMLHttpRequest */
 
-/* global cygnus, Blob, Worker, XMLHttpRequest */
+const _arguments = arguments
 
 module.exports = {
   supportsHistory: !!window.history,
   supportsWorkers: !!window.Worker,
   supportsPromises: !!Promise,
   ready: false,
   pages: {},
-  init: function() {
-
-    var _this2 = this;
-    if (!_this2.ready) {
-      window.cygnus = _this2;
-      window.onpopstate = _this2.handlePopState;
-      _this2.ready = true;
+  init: function () {
+    // Exit if history api, workers and promises aren't all supported
+    if (!this.supportsHistory || !this.supportsWorkers || !this.supportsPromises) {
+      console.info('[Cygnus]: cygnus is not supported in this browser.')
+      return false
     }
 
-    // Exit if history api, workers and promises aren't all supported
-    if (!_this2.supportsHistory || !_this2.supportsWorkers || !_this2.supportsPromises) {
-      console.info('[Cygnus]: cygnus is not supported in this browser.');
-      return false;
+    if (!this.ready) {
+      window.cygnus = this
```

Comment:
Agree... There is a potentially a bit of work to be done to make everything properly self-contained so that scope is always right to make it so that attaching to `window` isn't necessary, but I think this is probably the right way to go.